crates/astria-sequencer/src/accounts/ics20_transfer.rs

```diff
@@ -319,10 +319,8 @@ async fn execute_ics20_transfer<S: StateWriteExt>(
     dest_channel: &ChannelId,
     is_refund: bool,
 ) -> Result<()> {
-    use prost::Message as _;
-
-    let packet_data = FungibleTokenPacketData::decode(data)
-        .context("failed to decode FungibleTokenPacketData")?;
+    let packet_data: FungibleTokenPacketData =
+        serde_json::from_slice(data).context("failed to decode FungibleTokenPacketData")?;
```

Comment:
Should try to avoid language specific types in error messages (because this might show up in observability/logs and not be very helpful).
```suggestion
        serde_json::from_slice(data).context("failed to decode fungible token packet as json")?;
```
Also consider using this: https://docs.rs/serde_path_to_error/latest/serde_path_to_error/